prow/io/option.go

```diff
@@ -23,17 +23,40 @@ import (
 	"google.golang.org/api/googleapi"
 )
 
+type WriterOption interface {
+	Apply(wo *WriterOptions)
+}
+
 // WriterOptions are options for the opener Writer method
 type WriterOptions struct {
```

Comment:
Is there a reason we can't just get rid of this class?

Reply:
How would you replace it?

Reply:
I'm fine with what is happening, but I don't see the point of the WriterOption interface.

Reply:
Ah got it. Thought it's about the struct. The original discussion is here: https://github.com/kubernetes/test-infra/pull/17618#discussion_r426279391

Before the current implementation I had 3 for loops to "aggregate" the WriterOptions so that was the reason to change it. I'm fine with both variants.

Reply:
Getting rid of the loops is great, but I don't see why that necessitates a pointless interface that has the single implementor of this interface as one of its arguments.

Reply:
Okay so possible variants:
* current implementation
* current implementation + remove the interface and keep the Apply method

Are there other ones?

@alvaroaleman WDYT?

Reply:
Yeah, I am fine with removing the interface

Reply:
Done